components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/IoTHubMessageSink.cs

```diff
@@ -67,11 +68,21 @@ public class IoTHubMessageSink : IMessageSink, IDisposable {
                     sw.Start();
 
                     try {
-                        if (messagesCount == 1) {
-                            await _clientAccessor.Client.SendEventAsync(messageObjects.First()).ConfigureAwait(false);
+                        if (string.IsNullOrEmpty(routingInfo)) {
+                            if (messagesCount == 1) {
+                                await _clientAccessor.Client.SendEventAsync(messageObjects.First()).ConfigureAwait(false);
+                            }
+                            else {
+                                await _clientAccessor.Client.SendEventBatchAsync(messageObjects).ConfigureAwait(false);
+                            }
                         }
                         else {
-                            await _clientAccessor.Client.SendEventBatchAsync(messageObjects).ConfigureAwait(false);
+                            if (messagesCount == 1) {
+                                await _clientAccessor.Client.SendEventAsync(routingInfo, messageObjects.First()).ConfigureAwait(false);
```

Comment:
I would argue that we should not use methods that accept `outputName` parameter here as that is not supported when `DeviceClient` is used. That means that OPC Publisher will not be able to send data when it is deployed using device credentials, which is often the case when running it outside of IoT Edge context. That is also the situation that we have when debugging locally, which would be very complicated with this change.

Reply:
fixed